### PR TITLE
:sparkles:  video upload modal #336

### LINKED ISSUE #336: 🗒️ Video Block Modal
## User stories

As a user I would like to see a modal similar to the one below when I attempt to attach a video so that I can be able to choose the size of the video I would like to upload.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Description
Currently how the video block works, is once the play icon is clicked it opens the file explorer to allow  the upload the video file.

In this feature instead of opening the file explorer on click, a modal with the design below should be implemented and open the file explorer when the thumbnail space is clicked.

```markdown
1. Navigate to the video block
1. I click on the play icon.
1. A modal window should appear
1. The modal window contains the following:
  1. [A title]
  2. [Two input fields: One a name input and Second a video input]
  3. [Radio button options with the size options for the video **Note** : One should be able to select only one option]
1. A button that applies the picked option, the video file and the name of the video to the video block object.
```

**Screenshots**

Screenshot 2023-03-01 at 16 19 47

### Acceptance Criteria

1. Implement design as is above
1. Modal should appear over clicked video block.

Please also note:

1. If a conversation about the issue I’m working on is happening,
the outcomes should be documented within that issue (not in the comments).

### PR DESCRIPTION
**Pull request by Beulah-Matt**
# Description

I have implemented a video modal with a 
-Title
-Name inputs
-A thumbnail image
-Radio check marks to select video resolution.

On clicking the play button  in the video block, you are redirected to this modal.
When you click on the thumbnail, a file explorer window is opened to select a video file  

Currently, you are able to see the video in the modal in place of the thumbnail and the URL is prepopulated under the video link input.

Fixes #336
## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

#Screenshot
[Screencast from 11-05-23 19:41:17.webm](https://github.com/italanta/elewa/assets/109944021/fa09fd3d-78bb-4fde-b622-debaee62f39a)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Beulah-Matt** [comment]
@LemmyMwaura , I have opened a new PR without the rebased files,  please see this.

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  
+    this.closeModal()
+  }
+
+  closeModal(): void {
+    this.dialogRef.close();
+  }
+
+  async processVideo(event: any) {
+    if (event.target.files && event.target.files[0]) {
```
Comment (njoroge-jane) [inline libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts:70]: Add comments here to clarify what is happening

**Comment by njoroge-jane** [comment]
@Beulah-Matt  Great Job!

A few things to work on for the general UI: 

- The color of the modal heading and the pixel options should match be as per the design
- The input for the name and the media input field should be of the same length even after the thumbnail has been uploaded.
- The spacing around the modal should be as per design

NB: Kindly attach a video of the entire uploading process.

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts (hunk 1 above)
Comment (Beulah-Matt) [reply libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts:70]: Cool, working on this, I will let you know when it is done @njoroge-jane

**Review by LemmyMwaura — APPROVED** [review]

**Review by ReaganCn — APPROVED** [review]

**Review by njoroge-jane — APPROVED** [review]

**Pull request merged**